src/utils/telemetry.js

```diff
@@ -0,0 +1,492 @@
+   * @param {boolean} [clear = false]
+   *        Whether to clear out the scalars after snapshotting.
+   */
+  snapshotEvents(dataset, clear) {}
+
+  /**
+   * Serializes the histograms from the given dataset to a JSON-style object.
+   * The returned structure looks like:
+   *   { process1: {name1: {histogramData1}, name2:{histogramData2}...}}
+   *
+   * Where histogramDataN has the following properties:
+   *   min - minimum bucket size
+   *   max - maximum bucket size
+   *   histogram_type - HISTOGRAM_EXPONENTIAL, HISTOGRAM_LINEAR,
+   *                    HISTOGRAM_BOOLEAN, HISTOGRAM_FLAG, HISTOGRAM_COUNT,
+   *                    or HISTOGRAM_CATEGORICAL
+   *   counts - array representing contents of the buckets in the histogram
+   *   ranges - array with calculated bucket sizes
+   *   sum - sum of the bucket contents
+   *
+   * @param {dataset}
+   *        DATASET_RELEASE_CHANNEL_OPTOUT or DATASET_RELEASE_CHANNEL_OPTIN.
+   * @param {boolean} [clear = false]
+   *        Whether to clear out the scalars after snapshotting.
+   */
+  snapshotHistograms(dataset, clear) {}
+
+  /**
+   * Serializes the keyed histograms from the given dataset to a JSON-style
+   * object.
+   *
+   * The returned structure looks like:
+   *   {
+   *     process1: {
+   *       name1: {
+   *         histogramData1
+   *       },
+   *       name2: {
+   *         histogramData2
+   *       },
+   *       ...
+   *     }
+   *   }
+   *
+   * @param {dataset}
+   *        DATASET_RELEASE_CHANNEL_OPTOUT or DATASET_RELEASE_CHANNEL_OPTIN.
+   * @param {boolean} [clear = false]
+   *        Whether to clear out the scalars after snapshotting.
+   */
+  snapshotKeyedHistograms(dataset, clear) {}
+
+  /**
+   * Serializes the scalars from the given dataset to a JSON-style object and
+   * resets them.
+   *
+   * The returned structure looks like:
+   *   {
+   *     "process": {
+   *       "category1.probe": 1,
+   *       "category1.other_probe": false,
+   *       ...
+   *     },
+   *     ...
+   *   }.
+   *
+   * @param {dataset}
+   *        DATASET_RELEASE_CHANNEL_OPTOUT or DATASET_RELEASE_CHANNEL_OPTIN.
+   * @param {boolean} [clear = false]
+   *        Whether to clear out the scalars after snapshotting.
+   */
+  snapshotScalars(dataset, clear) {}
+
+  /**
+   * Serializes the keyed scalars from the given dataset to a JSON-style object
+   * and resets them.
+   * The returned structure looks like:
+   *   {
+   *     "process": {
+   *       "category1.probe": {
+   *         "key_1": 2,
+   *         "key_2": 1,
+   *         ...
+   *       },
+   *       ...
+   *     },
+   *     ...
+   *   }
+   *
+   * @param {dataset}
+   *        DATASET_RELEASE_CHANNEL_OPTOUT or DATASET_RELEASE_CHANNEL_OPTIN.
+   * @param {boolean} [clear = false]
+   *        Whether to clear out the scalars after snapshotting.
+   */
+  snapshotKeyedScalars(dataset, clear) {}
+}
+
+export { Telemetry };
```

Comment:
can we add this ship to devtools-modules

https://github.com/devtools-html/devtools-core/tree/master/packages/devtools-modules/src

the benefits are:

1. any tools using the launchpad can use the shim
2. all the shims are in one place
3. the shims are easier to test

Reply:
Yup, that would make a lot of sense.